### PR TITLE
feat(obstacle_avoidance_planner): explicitly insert zero velocity

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description

In obstacle avoidance planner, a stop point is not explicitly inserted in an output trajectory.
Therefore the stop point changes a little bit comparing before and after obstacle avoidance planner.

In this PR, I inserted the stop point in the output trajectory.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by takayuki5168** [comment]
without this PR
image

with this PR
image

**Comment by takayuki5168** [comment]
(@TakaHoribe @tkimura4 If it's hard to review, just give me LGTM. I'll check if this PR works a lot.)

File: planning/obstacle_avoidance_planner/src/node.cpp [hunk 1]
```diff
@@ -927,6 +939,10 @@ ObstacleAvoidancePlanner::generateOptimizedTrajectory(
   // calculate trajectory with EB and MPT
   auto optimal_trajs = optimizeTrajectory(path, cv_maps);
 
+  // calculate velocity
+  // NOTE: Velocity is not considered in optimization.
+  calcVelocity(path.points, optimal_trajs.model_predictive_trajectory);
```
Comment (takayuki5168) [inline planning/obstacle_avoidance_planner/src/node.cpp:944]: just memo

If optimization fails, the previously optimized trajectory is used. When this optimization failure occurs while the stop line changes from red to green, the velocity in the previous trajectory is not updated from red to green, and the ego keeps stopping.
Therefore calcVelocity was moved after optimization to update velocity even if the optimization fails.

**Review by tkimura4 — APPROVED** [review]
LGTM

**Pull request merged**